patches/minecraft/net/minecraft/client/renderer/EntityRenderer.java.patch

```diff
@@ -103,7 +103,18 @@
                      if (this.field_78531_r.field_71439_g.func_70644_a(MobEffects.field_76439_r))
                      {
                          float f15 = this.func_180438_a(this.field_78531_r.field_71439_g, p_78472_1_);
-@@ -1101,6 +1109,10 @@
+@@ -971,7 +979,9 @@
+     private float func_180438_a(EntityLivingBase p_180438_1_, float p_180438_2_)
+     {
+         int i = p_180438_1_.func_70660_b(MobEffects.field_76439_r).func_76459_b();
+-        return i > 200 ? 1.0F : 0.7F + MathHelper.func_76126_a(((float)i - p_180438_2_) * (float)Math.PI * 0.2F) * 0.3F;
++        float brightness = i > 200 ? 1.0F : 0.7F + MathHelper.func_76126_a(((float)i - p_180438_2_) * (float)Math.PI * 0.2F) * 0.3F;
++        IBlockState iblockstate = ActiveRenderInfo.func_186703_a(Minecraft.func_71410_x().field_71441_e, p_180438_1_, p_180438_2_);
++        return net.minecraftforge.client.ForgeHooksClient.getNightVisionBrightness(this, p_180438_1_, iblockstate, p_180438_2_, brightness);
```

Comment:
I am still not a fan of the event every tick, however its client side so it wouldn't effect the server's performance. 
However I still think something as simple as a flag to disable the decay blink would do.
You also don't pass in the i value, so people will have to re-do the math to determine how long thats going to last. This seems like a value that would be highly used, to preventing people from needing to recalc it with it right there to pass in seems like a good idea.

Beyond that, why not move this hook outside this function. What if modders want to effect only the light level and not the fog renderer? What about decoupling from the nightvision potion effect. If you don't decouple why not just have the flag as mentioned on the potion instance? How about moving this entire function to the effect instance and leaving it up to whoever adds the potion effect to the player instead of everyone on the event bus?

Reply:
I figured a flag would never get pulled. As well wanted to offer the option to change the effect based on other sources, including blocks and entities in the world.

I'll make a change to pass in 'i' as that is a very good idea.

I could decouple but was focusing on one step at a time. Looking for the simplest solution that resulted in the fewest lines of change. While also allowing the changes to be used beyond just turning the effect off. As the whole reason for the PR was not to solve my own problem. Instead, a few developers asked me to make it for use in NV items, blocks, and entities.

That being said, if you want me to decouple I can make the changes :)